### ISSUE TITLE
failing on step "Run Gather"

### ISSUE THREAD

**Issue by hehouts** [dib-lab/genome-grist#189]
I am trying to run one single metagenome through grist, and my picklists are empty, but I dont know why, and its causing the Run Gather step to fail.

The metagenome I'm using is local, so I started from an `.abundtrim.fq.gz`
(here: `/home/hehouts/dynamic-duos-virome/grist/outputs.smp1_mgx/abundtrim/CSM79HGP.abundtrim.fq.gz`)

Does this mean that the prefetch step didnt work? Or that the signature is weird?
```
Starting prefetch sweep across databases.
Found 0 signatures via prefetch; now doing gather.
found less than 100.0 kbp in common. => exiting
for given picklist, found 0 matches to 0 distinct values
```

Also there is only a `sig.zip` and no `sig.gz`, so I either 
1) cant use zless to check out the signature or 
2) I should hypothetically be able to do that, and the signature is gibberish for some other reason

Error Message:
```
== This is sourmash version 4.3.0. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

picking column 'match_md5' of type 'md5prefix8' from 'outputs.smp1_mgx/gather/CSM79HGP.prefe
tch.csv'
loaded 0 distinct values into picklist.
selecting specified query k=31
loaded query: CSM79HGP... (k=31, DNA)
loaded 1 databases.
Starting prefetch sweep across databases.
Found 0 signatures via prefetch; now doing gather.
found less than 100.0 kbp in common. => exiting
for given picklist, found 0 matches to 0 distinct values
Waiting at most 180 seconds for missing files.
MissingOutputException in line 924 of /home/hehouts/genome-grist-dev/genome_grist/conf/Snake
file:
Job Missing files after 180 seconds:
outputs.smp1_mgx/gather/CSM79HGP.gather.csv
outputs.smp1_mgx/gather/CSM79HGP.matches.sig.zip
This might be due to filesystem latency. If that is the case, consider to increase the wait 
time with --latency-wait.
Job id: 4 completed successfully, but some output files are missing. 4
Removing output files of failed job sourmash_gather_wc since they might be corrupted:
outputs.smp1_mgx/gather/CSM79HGP.gather.out
Job failed, going on with independent jobs.
```

config looks like:
```
prevent_sra_download: true
samples:
- CSM79HGP

outdir: outputs.smp1_mgx

prefetch_memory: 20e9

# for mgx, when looking for bacteria
#sourmash_databases:
#- gtdb-rs202.genomic-reps.k31.sbt.zip

sourmash_compute_ksizes:
- 21
- 31
- 51

#prefetch_memory: 20e9

sourmash_databases:
- /group/ctbrowngrp/gtdb/databases/gtdb-rs202.genomic-reps.k31.zip

taxonomies:
- /group/ctbrowngrp/genbank/all_genbank_lineages.20200727.csv
```

still working on this, but if anyone has ideas on what issue I might be bumping into, please lmk :)

**Comment by ctb** [comment]
yay error message!!

On Thu, Jun 02, 2022 at 05:51:40PM -0700, Hannah Eve Houts wrote:
> I am trying to run one single metagenome through grist, and my picklists are empty, but I dont know why, and its causing the Run Gather step to fail.
> 
> The metagenome I'm using is local, so I started from an `.abundtrim.fq.gz`
> (here: `/home/hehouts/dynamic-duos-virome/grist/outputs.smp1_mgx/abundtrim/CSM79HGP.abundtrim.fq.gz`)
> 
> Does this mean that the prefetch step didnt work? Or that the signature is weird?
> ```
> Starting prefetch sweep across databases.
> Found 0 signatures via prefetch; now doing gather.
> found less than 100.0 kbp in common. => exiting
> for given picklist, found 0 matches to 0 distinct values
> ```

what threshold are you using for prefetch? 10 kb?

similar question for gather :)

> Also there is only a `sig.zip` and no `sig.gz`, so I either 
> 1) cant use zless to check out the signature or 
> 2) I should hypothetically be able to do that, and the signature is gibberish for some other reason

try:

unzip <filename>.sig.zip

will dig more tomorrow!

**Comment by ctb** [comment]
Took a look today - it seems like maybe it's just a small metagenome with no matches above threshold.

I used:
```
sourmash sig describe CSM79HGP.abundtrim.sig.zip
```
to check the size of the metagenome signature - it reported
```
signature filename: /home/hehouts/dynamic-duos-virome/grist/outputs.smp1_mgx/sigs/CSM79HGP.abundtrim.sig.zip
signature: CSM79HGP
source file: outputs.smp1_mgx/abundtrim/CSM79HGP.abundtrim.fq.gz
md5: c745ab8cd32176ecba3d966687761b19
k=21 molecule=DNA num=0 scaled=1000 seed=42 track_abundance=1
size: 2136
sum hashes: 239653
signature license: CC0
```

which ... now that I think about it, looks a bit odd. Specifically, you have few distinct k-mers (2136 hashes, so about 2 Mbp (2136 x scaled=1000) of sequence, but then the abundances of those hashes is very large (sum_hashes=239653, divide by size=2136 => ~1000 average abundance of each hash!)

in sum: all looks correct, "just" a weird sample.

looks like it's a metatranscriptome - https://www.ebi.ac.uk/ena/browser/text-search?query=CSM79HGP - maybe that explains it?

**Comment by hehouts** [comment]
> looks like it's a metatranscriptome - https://www.ebi.ac.uk/ena/browser/text-search?query=CSM79HGP - maybe that explains it?
😭
Ok, Im going to figure out how to get these samples from SRA. Because the file manifest clearly says this is a metagenome, so I wonder if many of the other metagenomes are actually transcriptomes.

**Comment by ctb** [comment]
> Ok, Im going to figure out how to get these samples from SRA. Because the file manifest clearly says this is a metagenome, so I wonder if many of the other metagenomes are actually transcriptomes.

What do you mean "get these samples from the SRA"? genome-grist downloads them from the SRA.

And where is "the file manifest" coming from - the iHMP folk? We can ask them, too.

**Comment by hehouts** [comment]
I used metagenomes that were downloaded onto farm from the iHMP portal. So for grist I used `prevent_sra_download: true` and started with `abundtrim.fq.gz` files.
The manifest Im using comes from [IBDMDB website](https://ibdmdb.org/tunnel/public/summary.html), and I needed it to [connect file name with patient & time point](https://github.com/hehouts/dynamic-duos/blob/master/R/metadata_summary.md#chunk-5---s5-subset-file-ids-for-viromes), and then with associated viromes. 

[I considered downloading these ](https://hackmd.io/y9sCKPjnQD-e1S6x1kQstQ?view#Where-samples-are-available-for-download) samples from the SRA a few months ago, but wound up dropping that plan in favor of downloading from the iHMP portal because I had patient and time point info.

So I think I will try running grist on a few metagenomes, starting with accession numbers, and work on trying to figure out what the sample patient & time point info is later. 

I previously wanted to try to use the IBDMDB metadata table but couldnt find a way to merge it with SRA accession numbers. But now I am suspicious of that metadata, so this doesnt seem as important
